clients/da-vinci-client/src/main/java/com/linkedin/davinci/validation/KafkaDataIntegrityValidator.java

```diff
@@ -24,30 +29,53 @@
  * 4. Whether producers have produced duplicate messages, which is fine and expected due to producer retries (DUPLICATE).
  */
 public class KafkaDataIntegrityValidator {
+  private static final Logger LOGGER = LogManager.getLogger(KafkaDataIntegrityValidator.class);
+  public static final long DISABLED = -1;
+  private final String versionTopicName;
   private final long kafkaLogCompactionDelayInMs;
+  private final long maxAgeInMs;
+  /** The oldest producer timestamp we retained during the last run of {@link #clearExpiredState()} */
+  private volatile long oldestRetainedProducerTS = 0;
 
   /** Keeps track of every upstream producer this consumer task has seen so far. */
   protected final Map<GUID, ProducerTracker> producerTrackerMap;
   protected final Function<GUID, ProducerTracker> producerTrackerCreator;
+  private final Time time;
 
-  public KafkaDataIntegrityValidator(String kafkaVersionTopic) {
-    this(kafkaVersionTopic, -1);
+  public KafkaDataIntegrityValidator(String versionTopicName, long kafkaLogCompactionDelayInMs) {
```

Comment:
optional/maybe nit?: Does this class only work for version topics or can it be used for RT?  If not, maybe just remove the version notion so folks don't make that assumption later?

Reply:
IIUC, the topic name is there only for logging purposes, and the class ought to work for RT (or any Venice topics, really)... I'll change the variable name. Thanks for pointing it out!